src/frontend/views/view_utils.py

```diff
@@ -0,0 +1,41 @@
+
+from django.shortcuts import get_object_or_404
+from src.apps.runs.models import Run
+
+def set_current_run_or_run_from_url_for_view_get_queryset(view):
+  """Helper for implementing get_queryset in a view that cares about the run.
+
+  Sets view.run to:
+    * The run specified by the url (from view.kwargs)
+    * Or else, the current run if there is a current active run
+    * Or else, None
+
+  Sets view.viewing_current_run to False if the run was in the url, else True.
+  """
```

Comment:
Also I start being less clear about this.

- having no run in url should means: "I want to see data from all runs" (eg my overall contributions).
- else we should always have a run in url, explicitly.

Reply:
I left it this way for networks since for networks it make a lot of sense to default to current run. But for contributors I took your suggestion and greatly simplified to to just view all games from all runs. Thanks!